src/unix/fs.c

```diff
@@ -188,9 +193,38 @@ static ssize_t uv__fs_futime(uv_fs_t* req) {
   }
 
   return r;
+#elif defined(__APPLE__)
+  struct attrlist attr_list;
+  struct timespec times[3];
+  unsigned i = 0;
+
+  bzero(&attr_list, sizeof(attr_list));
+
+  attr_list.bitmapcount = ATTR_BIT_MAP_COUNT;
+
+  if (!isnan(req->btime)) {
+    attr_list.commonattr |= ATTR_CMN_CRTIME;
+
+    times[i++].tv_sec = req->btime;
+    times[i].tv_nsec = (unsigned long)(req->btime * 1000000) % 1000000 * 1000;
+  }
+
+  if (!isnan(req->atime)) {
+    attr_list.commonattr |= ATTR_CMN_ACCTIME;
+
+    times[i++].tv_sec = req->atime;
+    times[i].tv_nsec = (unsigned long)(req->atime * 1000000) % 1000000 * 1000;
+  }
 
-#elif defined(__APPLE__)                                                      \
-    || defined(__DragonFly__)                                                 \
+  if (!isnan(req->mtime)) {
+    attr_list.commonattr |= ATTR_CMN_MODTIME;
+
+    times[i++].tv_sec = req->mtime;
+    times[i].tv_nsec = (unsigned long)(req->mtime * 1000000) % 1000000 * 1000;
+  }
+
+  return fsetattrlist(req->file, &attr_list, times, i * sizeof(times), 0);
```

Comment:
Please use `i * sizeof(times[0])` in that case.  It doesn't really matter here (apart from being libuv convention) but in general it's better to take the size of variables rather than types because the type of a variable can change.